tests/platform_tests/fwutil/fwutil_common.py

```diff
@@ -0,0 +1,228 @@
+        logger.info("Waiting on switch to come up....")
+        localhost.wait_for(host=hn, port=22, state='started', delay=10, timeout=150)
+        wait_until(300, 30, duthost.critical_services_fully_started)
+        time.sleep(60)
+
+def show_firmware(duthost):
+    out = duthost.command("fwutil show status")
+    
+    num_spaces = 2
+    curr_chassis = ""
+    output_data = {"chassis":{}}
+    status_output = out['stdout']
+    separators = re.split(r'\s{2,}', status_output.splitlines()[1])  # get separators
+    output_lines = status_output.splitlines()[2:]
+
+    for line in output_lines:
+        data = []
+        start = 0
+
+        for sep in separators:
+            curr_len = len(sep)
+            data.append(line[start:start+curr_len].strip())
+            start += curr_len + num_spaces
+
+        if data[0].strip() != "":
+            curr_chassis = data[0].strip()
+            output_data["chassis"][curr_chassis] = {"component": {}}
+
+        output_data["chassis"][curr_chassis]["component"][data[2]] = data[3]
+
+    return output_data
+
+def get_install_paths(duthost, fw, versions, chassis):
+    chass = show_firmware(duthost)["chassis"].keys()[0]
+    component = fw["chassis"].get(chass, {})["component"]
+    ver = versions["chassis"].get(chass, {})["component"]
+    
+    paths = {}
+
+    for comp, revs in component.items():
+        if comp in ver:
+            if revs[0].get("upgrade_only", False) and ver[comp] not in [r["version"] for r in revs]:
+                # If upgrade only we need to skip if the installed version isn't on record
+                continue
+            for i, rev in enumerate(revs):
+                if rev["version"] != ver[comp]:
+                    paths[comp] = rev
+                    break
+                elif rev.get("upgrade_only", False):
+                    # If upgrade only stop searching when installed version hit
+                    break
+    return paths
+
+def generate_config(duthost, cfg, versions):
+    valid_keys = ["firmware", "version"]
+    chassis = versions["chassis"].keys()[0]
+    paths = deepcopy(cfg)
+
+    # Init all the components to null
+    for comp in versions["chassis"][chassis]["component"].keys():
+        paths[comp] = paths.get(comp, {})
+        if "firmware" in paths[comp]:
+            paths[comp]["firmware"] = os.path.join("/", DEVICES_PATH, 
+                    duthost.facts["platform"], 
+                    os.path.basename(paths[comp]["firmware"]))
+
+    # Populate items we are installing
+    with open("platform_components.json", "w") as f:
+        json.dump({"chassis":{chassis:{"component":{comp:{k: v 
+            for k, v in dat.items() 
+            if k in valid_keys} 
+            for comp, dat in paths.items()}}}}, f, indent=4)
+
+def upload_platform(duthost, paths, next_image=None):
+    target = next_image if next_image else "/"
+
+    # Copy over the platform_components.json file
+    duthost.copy(src="platform_components.json", 
```

Comment:
@alexrallen do we have a logic to restore the original platform_components.json once the testing is done?

Reply:
Added.